src/main.rs

```diff
@@ -53,89 +67,113 @@ fn main_loop(mut surface: GL33Surface) {
         angle_drag     = [0.0, 2.0]    random::TwoDice,
     );
 
-    let start_t = Instant::now();
-
-    let mut input = input::InputManager::new(surface.sdl());
-    input.bind(input::Device::Key(input::Keycode::A), Name::Left);
-    input.bind(input::Device::Key(input::Keycode::D), Name::Right);
-    input.bind(input::Device::Key(input::Keycode::W), Name::Up);
-    input.bind(input::Device::Key(input::Keycode::S), Name::Down);
-    input.bind(input::Device::Key(input::Keycode::Escape), Name::Quit);
-    input.bind(input::Device::Key(input::Keycode::F), Name::PlaySound);
-    input.bind(input::Device::Quit, Name::Quit);
-    input.bind(input::Device::Axis(0, input::Axis::LeftX), Name::Right);
-    input.bind(input::Device::Axis(0, input::Axis::RightY), Name::Up);
-
-    let mut audio = Audio::init(surface.sdl());
-    audio.resume();
-
-    let bloop = assets.load_audio(Path::new("res/bloop.wav").to_path_buf());
-
-    let mut old_t = start_t.elapsed().as_millis() as f32 * 1e-3;
-    'app: loop {
-        let t = start_t.elapsed().as_millis() as f32 * 1e-3;
-        let delta = t - old_t;
-        old_t = t;
-
-        performance::frame();
-
+    'main: loop {
+        // Go a step forward in time.
+        let delta = game.time_tick();
+        game.update(delta);
+        //XXX Here we could choose to go the other way and have our game struct take an update and
+        //draw function. But what happens with global state then? Maybe we can do:
+        /*
+         * pub struct Game<STATE> {
+         *     // ...
+         *     state: STATE,
+         *     update: fn(&mut STATE, f32),
+         *     draw: fn(&mut STATE),
+         * }
+         *
+         * impl<S> Game<S> {
+         *     fn run(&mut self) {
+         *         loop {
+         *             // ...
+         *             update(&mut self.state, delta);
+         *             // ...
+         *             draw(&mut self.state);
+         *             // ...
+         *         }
+         *     }
+         * }
+         */
```

Comment:
How would this work with sylt? The bindings will require the context when they're running. We can always ignore the argument passed to the functions, but it makes our jobb harder...